packages/core/state/selection/actions.ts

```diff
@@ -75,6 +78,187 @@ export function removeFileFilter(filter: FileFilter | FileFilter[]): RemoveFileF
     };
 }
 
+/**
+ * SET_FUZZY_FILTERS
+ *
+ * Intention to set, wholesale, a list of FuzzyFilters into application state. This should not be dispatched
+ * by UI components; dispatch either an ADD_FUZZY_FILTER or REMOVE_FUZZY_FILTER action. Those actions will
+ * trigger the `modifyFuzzyFilters` logic, which will then dispatch this action.
+ */
+export const SET_FUZZY_FILTERS = makeConstant(STATE_BRANCH_NAME, "set-fuzzy-filters");
```

Comment:
This is another instance where there is so much similarity between fuzzy include and exclude. Is there a considerable refactor that could happen here? seeing a lot of code in triplicate.

Reply:
Absolutely, I'd ideally like to remove all the duplicate code and streamline the different filter types. It might be worth making it a separate refactor ticket though, since this feature has been open for a long time now (especially fuzzy search)